pkg/controllers/dynakube/version/reconciler.go

```diff
@@ -53,13 +54,17 @@ func (r *reconciler) ReconcileOneAgent(ctx context.Context, dynakube *dynatracev
 }
 
 func (r *reconciler) ReconcileActiveGate(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) error {
-	updaters := []StatusUpdater{
-		newActiveGateUpdater(dynakube, r.apiReader, r.dtClient),
+	if !dynakube.NeedsActiveGate() {
+		meta.RemoveStatusCondition(dynakube.Conditions(), activeGateVersionConditionType)
+
+		return nil
 	}
```

Comment:
in all other cases we have the this logic in the `updater.Enabled` function

Reply:
Already doing it there, thanks for pointing out 👍